pkg/descheduler/evictions/evictions.go

```diff
@@ -188,53 +191,64 @@ func (pe *PodEvictor) Evictable(opts ...func(opts *Options)) *evictable {
 	}
 
 	ev := &evictable{}
+	if !pe.evictSystemCriticalPods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			// Moved from IsEvictable function to allow for disabling
+			if utils.IsCriticalPriorityPod(pod) {
+				return fmt.Errorf("pod has system critical priority")
+			}
+			return nil
+		})
+	}
 	if !pe.evictLocalStoragePods {
 		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if IsPodWithLocalStorage(pod) {
-				return fmt.Errorf("pod has local storage and descheduler is not configured with --evict-local-storage-pods")
+			if utils.IsPodWithLocalStorage(pod) {
+				return fmt.Errorf("pod has local storage and descheduler is not configured with evictLocalStoragePods")
 			}
 			return nil
 		})
 	}
 	if pe.ignorePvcPods {
 		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if IsPodWithPVC(pod) {
+			if utils.IsPodWithPVC(pod) {
 				return fmt.Errorf("pod has a PVC and descheduler is configured to ignore PVC pods")
 			}
 			return nil
 		})
 	}
-	if options.priority != nil {
+	if !pe.evictSystemCriticalPods && options.priority != nil {
```

Comment:
Worth mention that `Priority filtering` won't work if `evictSystemCriticalPods` is true

Reply:
I'll add that to the README.md